### PR TITLE
[BLAS] add optional index_base parameter to iamax, iamin

### PR DESCRIPTION
**Pull request by andrewtbarker**
Adds optional index_base parameter so iamax and iamin can return 1-based indices if user desires.

### CODE REVIEW

File: source/elements/oneMKL/source/domains/blas/iamax.rst [hunk 1]
```diff
@@ -165,6 +167,10 @@ iamax (USM Version)
    incx
       The stride of vector ``x``.
 
+   base
```
Comment (mmeterel) [inline source/elements/oneMKL/source/domains/blas/iamax.rst:170]: Looks like this is missing in Syntax for USM Version.

File: source/elements/oneMKL/source/domains/blas/iamax.rst [hunk 2]
```diff
@@ -59,7 +55,8 @@ iamax (Buffer Version)
                   std::int64_t n,
                   sycl::buffer<T,1> &x,
                   std::int64_t incx,
-                  sycl::buffer<std::int64_t,1> &result)
+                  sycl::buffer<std::int64_t,1> &result,
+                  index_base base = index_base::zero)
```
Comment (mmeterel) [inline source/elements/oneMKL/source/domains/blas/iamax.rst:59]: I think the standard is to add the namespace for the types (like onemkl::transpose transa in gemm)

File: source/elements/oneMKL/source/domains/blas/iamax.rst (hunk 1 above)
Comment (andrewtbarker) [reply source/elements/oneMKL/source/domains/blas/iamax.rst:172]: Fixed, thanks!

**Review by mmeterel — APPROVED** [review]

**Comment by andrewtbarker** [comment]
Thanks Mesut, please go ahead and merge, I don't have permission on this repo.

**Pull request merged**